Clarify Kubernetes Getting Started guide

ptgott opened the pull request:
This change address some issues I found while working through
the "Getting Started - Kubernetes with SSO" guide.

- Assign required env variables in code snippets, and offer
  instructions for changing them. Otherwise, a reader may not
  realize that these variables are required for a command when
  pasting code.

- Put the AWS Route 53 jq command on one line. This way, users
  can copy it using our single-line copy button, and most of
  the command will no longer display as a comment.

- Add an admonition that describes the difference between this
  guide and our other Kubernetes Getting Started guide. This
  way, a reader won't have to read through that guide to
  understand why we have two Getting Started guides for
  Kubernetes.

- Indicate that a domain name is required. Not having one
  really tripped me up when I worked through this guide the
  first time.

- For users with environments that preclude access to the public
  internet, I added an admonition linking to the docker-compose
  guide and our contact page, with a mention of our on-prem
  Kubernetes support. We can replace this with a link to a
  local-only Kubernetes guide when one is available.

- While some Kubernetes load balancers populate the hostname,
  rather than the IP in the status field, this guide only
  includes instructions for fetching the IP. This adds
  instructions for fetching the hostname as well.

  See:
  https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#ServiceStatus

Closes #9086
Closes #9112

ptgott committed:
Clarify Kubernetes Getting Started guide

This change address some issues I found while working through
the "Getting Started - Kubernetes with SSO" guide.

- Assign required env variables in code snippets, and offer
  instructions for changing them. Otherwise, a reader may not
  realize that these variables are required for a command when
  pasting code.

- Put the AWS Route 53 jq command on one line. This way, users
  can copy it using our single-line copy button, and most of
  the command will no longer display as a comment.

- Add an admonition that describes the difference between this
  guide and our other Kubernetes Getting Started guide. This
  way, a reader won't have to read through that guide to
  understand why we have two Getting Started guides for
  Kubernetes.

- Indicate that a domain name is required. Not having one
  really tripped me up when I worked through this guide the
  first time.

- For users with environments that preclude access to the public
  internet, I added an admonition linking to the docker-compose
  guide and our contact page, with a mention of our on-prem
  Kubernetes support. We can replace this with a link to a
  local-only Kubernetes guide when one is available.

- While some Kubernetes load balancers populate the hostname,
  rather than the IP in the status field, this guide only
  includes instructions for fetching the IP. This adds
  instructions for fetching the hostname as well.

  See:
  https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#ServiceStatus

Closes #9086
Closes #9112

zmb3 commented on docs/pages/kubernetes-access/getting-started/cluster.mdx:
```diff
@@ -5,12 +5,18 @@ description: Getting started with Teleport. Let's deploy Teleport in a Kubernete
 # Getting Started
 
-Let's deploy Teleport in a Kubernetes with SSO and Audit logs:
+Let's deploy Teleport in a Kubernetes cluster with SSO and Audit logs:
```
Readers might get a little bit confused that we're using the same kubernetes cluster as both:

- a place to deploy teleport
- the resource that teleport will provide access to

Can we try to clarify this a bit? Maybe saying something like:

"Teleport can provide secure, unified access to your Kubernetes clusters. This guide assumes you don't have an existing Teleport cluster running, and will deploy Teleport on the same Kubernetes cluster you'll provide access to."

zmb3 commented on docs/pages/kubernetes-access/getting-started/cluster.mdx:
````diff
@@ -23,18 +29,29 @@ Let's deploy Teleport in a Kubernetes with SSO and Audit logs:
 <Tabs>
   <TabItem label="Open Source">
     ```code
+    $ CLUSTER_NAME="tele.example.com"
````
```suggestion
    $ export CLUSTER_NAME="tele.example.com"
```

Export is required in order for `CLUSTER_NAME` to remain set for future commands.

ptgott replied:
Since `CLUSTER_NAME` and `EMAIL` are only used in the initial shell environments of the two code snippets, and their values are passed to the `helm install` command via parameter expansion (i.e. the variables themselves aren't part of the environment of `helm install`), would it be safer to leave these variables unexported?

ptgott committed:
Respond to PR feedback and add edits

- Clarify that the Kubernetes cluster we deploy on here is also
  the one we enable access to.

- Minor readability edits

r0mant commented on docs/pages/kubernetes-access/getting-started/cluster.mdx:
```diff
@@ -23,18 +28,29 @@ Let's deploy Teleport in a Kubernetes with SSO and Audit logs:
+<Admonition title="Local-only setups" type="tip">
+Teleport also supports Kubernetes in on-premise and air-gapped environments. If you would like to try out Teleport on your local machine, we recommend following our [Docker Compose guide](../../getting-started/docker-compose.mdx). 
+
+You can also [contact us](/contact-us/) so we can work with you to determine the best setup for your Kubernetes environment.
```
I'd remove this sentence TBH. Seems irrelevant to the guide.

r0mant commented on docs/pages/kubernetes-access/getting-started/cluster.mdx:
````diff
@@ -101,6 +125,12 @@ to create a public IP for Teleport.
+    If `$MYIP` is blank, your cloud provider may have assigned a hostname to the load balancer rather than an IP address. Run the following command to retrieve the hostname:
+
+    ```code 
+    $ MYHOSTNAME=$(kubectl get services teleport-cluster -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
````
How is `MYHOSTNAME` used? It doesn't seem to be used below, only `MYIP` is.

ptgott committed:
Respond to PR feedback

- Remove superfluous sentence
- Assign MYIP to the load balancer's hostname if the IP is not
  available, since we use this variable in subequent commands.
- Fix linter issues
